kuma/core/models.py

```diff
@@ -34,6 +38,22 @@ def update(self, **kw):
                                           created=False)
 
 
+class IPBan(models.Model):
+    ip = models.GenericIPAddressField()
+    created = models.DateTimeField(default=timezone.now, db_index=True)
+    deleted = models.DateTimeField(null=True, blank=True)
+
+    objects = IPBanManager()
+
+    def delete(self, *args, **kwargs):
+        self.deleted = timezone.now()
```

Comment:
All the ones I saw in https://www.djangopackages.com/grids/g/deletion/ looked either too heavyweight or unstable/un-maintained. :frowning: Unless one of them looks right to you?